azurerm/internal/services/network/express_route_circuit_connection_resource.go

```diff
@@ -177,13 +177,11 @@ func resourceExpressRouteCircuitConnectionRead(d *pluginsdk.ResourceData, meta i
 		}
 		d.Set("address_prefix_ipv6", addressPrefixIPv6)
 
-		if props.PeerExpressRouteCircuitPeering != nil && props.PeerExpressRouteCircuitPeering.ID != nil {
-			circuitPeerPeeringId, err := parse.ExpressRouteCircuitPeeringID(*props.PeerExpressRouteCircuitPeering.ID)
-			if err != nil {
-				return err
-			}
-			d.Set("peer_peering_id", circuitPeerPeeringId.ID())
+		circuitPeerPeeringId, err := parse.ExpressRouteCircuitPeeringID(*props.PeerExpressRouteCircuitPeering.ID)
```

Comment:
Wondering why the omit the nil check here?

Reply:
Sorry. I didn't get your point. https://github.com/terraform-providers/terraform-provider-azurerm/pull/11303#discussion_r654242616